test/test_images/latencymako/aggregator.go

```diff
@@ -30,49 +27,59 @@ import (
 	"github.com/golang/protobuf/ptypes"
 	"github.com/golang/protobuf/ptypes/timestamp"
 
-	pb "knative.dev/eventing/test/test_images/latencymako_aggregator/event_state"
-	"knative.dev/pkg/signals"
+	pb "knative.dev/eventing/test/test_images/latencymako/event_state"
 	"knative.dev/pkg/test/mako"
 )
 
 const listenAddr = ":10000"
 const maxRcvMsgSize = 1024 * 1024 * 10
 
-// flags for the image
-var (
-	verbose       bool
-	expectRecords uint
-)
-
-var fatalf = log.Fatalf
-
-func init() {
-	flag.BoolVar(&verbose, "verbose", false, "Enable verbose logging")
-	flag.UintVar(&expectRecords, "expect-records", 1, "Number of expected events records before aggregating data.")
-}
-
-// aggregation of received events records
+// thread-safe events recording map
 type eventsRecord struct {
 	sync.RWMutex
-	pb.EventsRecord
+	*pb.EventsRecord
 }
 
-var (
-	sentEvents     eventsRecord
-	acceptedEvents eventsRecord
-	failedEvents   eventsRecord
-	receivedEvents eventsRecord
-)
+type aggregatorExecutor struct {
+	// thread-safe events recording maps
+	sentEvents     *eventsRecord
+	acceptedEvents *eventsRecord
+	failedEvents   *eventsRecord
+	receivedEvents *eventsRecord
 
-var notifyEventsReceived = make(chan struct{})
+	// channel to notify the main goroutine that an events record has been received
+	notifyEventsReceived chan struct{}
+}
 
-func main() {
-	flag.Parse()
+func newAggregatorExecutor() testExecutor {
```

Comment:
Instead of doing everything inside a single main, executor fields (chans, maps, ...) are initialized in separate `new` funcs.